lint/golangci-lint.bzl

```diff
@@ -0,0 +1,146 @@
+"""API for declaring a golangci-lint lint aspect that visits go_library, go_test, and go_binary rules.
+
+```
+load("@aspect_rules_lint//lint:golangci-lint.bzl", "golangci_lint_aspect")
+
+golangci_lint = golangci_lint_aspect(
+    binary = "@@//tools:golangci-lint",
+    config = "@@//:.golangci.yaml",
+)
+```
+"""
+
+load("@bazel_skylib//rules:native_binary.bzl", "native_binary")
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+load("@io_bazel_rules_go//go:def.bzl", "go_context")
+load("//lint/private:lint_aspect.bzl", "report_file")
+
+_MNEMONIC = "golangcilint"
+
+def golangci_lint_action(ctx, executable, srcs, config, report, use_exit_code = False):
+    """Run golangci-lint as an action under Bazel.
+
+    Based on https://github.com/golangci/golangci-lint
+
+    Args:
+        ctx: Bazel Rule or Aspect evaluation context
+        executable: label of the the golangci-lint program
+        srcs: golang files to be linted
+        config: label of the .golangci.yaml file
+        report: output file to generate
+        use_exit_code: whether to fail the build when a lint violation is reported
+    """
+
+    # golangci-lint calls out to Go, so we need the go context
+    go = go_context(ctx)
+    inputs = srcs + [config] + go.sdk_files
+    args = ctx.actions.args()
+    args.add_all(srcs)
+
+    command = """#!/usr/bin/env bash
+        export GOROOT=$(cd "$(dirname {go_tool})/.."; pwd)
```

Comment:
would be nice to link to docs here as well, just in case rules_go changes something that requires users to account for it like adding another variable